sdk/core/Azure.Core.Experimental/tests/public/JsonDataPublicTests.cs

```diff
@@ -310,17 +300,71 @@ public void CanCastToIEnumerableOfT()
         }
 
         [Test]
-        [Ignore("To be implemented.")]
         public void EqualsHandlesStringsSpecial()
         {
             dynamic json = new BinaryData("\"test\"").ToDynamicFromJson();
 
             Assert.IsTrue(json.Equals("test"));
-            Assert.IsTrue(json.Equals(new BinaryData("\"test\"").ToDynamicFromJson()));
+            Assert.IsTrue(json == "test");
+        }
+
+        [Test]
+        [TestCase("\"test\"", "\"test\"", true)]
+        [TestCase("1", "1.0", true)]
+        [TestCase("5.5", "5.5", true)]
+        [TestCase("true", "true", true)]
+        [TestCase("false", "false", true)]
+        [TestCase("null", "null", true)]
+        [TestCase("\"test\"", "\"wrong\"", false)]
+        [TestCase("\"test\"", "1", false)]
+        [TestCase("true", "false", false)]
+        [TestCase("1.1", "1.2", false)]
+        [TestCase("1", "1.2", false)]
+        [TestCase("1", "2", false)]
+        [TestCase("1", "null", false)]
+        [TestCase("1", "{ \"foo\": 1 }", false)]
+        public void EqualsPrimitiveValues(string a, string b, bool expected)
```

Comment:
We actually already have those in this section: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.Experimental/tests/public/JsonDataPublicTests.cs#L341

Those were functional prior to this change because the cast operators were present on DynamicData.

It makes sense we could do more to connect those to make that clear -- I can add that in the next organizational pass over the tests.  Thanks!